arch/x86/kvm/pkvm/pgtable.c

```diff
@@ -141,3 +312,19 @@ int pkvm_pgtable_walk(struct pkvm_pgtable *pgt, unsigned long vaddr,
 	root = mm_ops->phys_to_virt(pgt_ops->pte_to_phys(&pgt->root));
 	return _pgtable_walk(&data, root, pgt->cap.level);
 }
+
+int pkvm_pgtable_mmu_map(unsigned long vaddr, unsigned long phys,
```

Comment:
Why have specific functions pkvm_pgtable_mmu_map() and pkvm_pgtable_host_mmu_map() instead of a single generic function pkvm_pgtable_map() like in the existing implementation?

If we really want dedicated *_map() functions for hyp/host/guest MMU, they can be just wrappers around pkvm_pgtable_map(), and then they can be implemented in mmu.c, not here in the generic pgtable framework.

I know pKVM-ARM's pgtable framework has separate functions kvm_pgtable_hyp_map() and kvm_pgtable_stage2_map(). But our pgtable structure is already more abstract than ARM's one, so why not take advantage of that.

Reply:
Yeah, SGTM.